server/src/test/java/org/elasticsearch/index/mapper/ObjectMapperMergeTests.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.mapper;
+
+import com.google.common.collect.ImmutableMap;
+import org.elasticsearch.common.Explicit;
+import org.elasticsearch.common.joda.FormatDateTimeFormatter;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.mapper.FieldMapper.CopyTo;
+import org.elasticsearch.index.mapper.FieldMapper.MultiFields;
+import org.elasticsearch.index.mapper.TextFieldMapper.TextFieldType;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.AfterClass;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.cluster.metadata.IndexMetaData.SETTING_VERSION_CREATED;
+import static org.elasticsearch.index.mapper.ObjectMapper.Dynamic;
+import static org.elasticsearch.index.mapper.ObjectMapper.Nested;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class ObjectMapperMergeTests extends ESTestCase {
+
+    private static FieldMapper barFieldMapper = createTextFieldMapper("bar");
+    private static FieldMapper bazFieldMapper = createTextFieldMapper("baz");
+
+    private static RootObjectMapper rootObjectMapper = createRootObjectMapper(
+        "type1", true, ImmutableMap.of(
+            "disabled", createObjectMapper("disabled", false, emptyMap()),
+            "foo", createObjectMapper("foo", true, ImmutableMap.of(
+                "bar", barFieldMapper))));
+
+    private static Explicit<Boolean> dateDetection = new Explicit<>(false, false);
+    private static Explicit<Boolean> numericDetection = new Explicit<>(false, false);
+    private static Explicit<FormatDateTimeFormatter[]> dynamicDateTimeFormatters = new Explicit<>(new FormatDateTimeFormatter[0], false);
+    private static Explicit<DynamicTemplate[]> dynamicTemplates = new Explicit<>(new DynamicTemplate[0], false);
```

Comment:
I was a bit suprised by these until I saw them being used in createRootObjectMapper(), maybe you could just instantiate them there locally, even if that means a few more instances. I think it makes the test more readable although this way is certainly "better" in terms of object creation.

Reply:
Yes, you're right. It would be a better balance between performances and readability. I'll do it, thank you.